### PR TITLE
[data_processing][rne] add DAG to download RNE stock data and get daily flux from API

### LINKED ISSUE #92: [DATA] Télécharegr les fichiers json stock et flux API RNE

### PR DESCRIPTION
**Pull request by HAEKADI**
closes #92 
This PR adds two DAGs:
* **Stock DAG** : fetches a zip file from the INPI website, processes the JSON files within it, and saves them to a MinIP instance. These files will subsequently be used to generate the `stock` files required for `dirigeants` data in `Annuaire des Entreprises`.
* **Flux DAG** : retrieves data for SIREN numbers that have been modified since the last modification date saved in JSON files on MinIO, up until the day before the execution. These data are then stored on a Minio instance. These files will subsequently be used to update `Annuaire des Entreprises` search API.

### CODE REVIEW

File: data_processing/rne/task_functions.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+DAG_FOLDER = "datagouvfr_data_pipelines/data_processing/"
+TMP_FOLDER = f"{AIRFLOW_DAG_TMP}rne/"
+DATADIR = f"{TMP_FOLDER}data"
+ZIP_FILE_PATH = f"{TMP_FOLDER}rne.zip"
+EXTRACTED_FILES_PATH = f"{TMP_FOLDER}extracted/"
+
+
+def get_rne_stock():
+    download_files(
```
Comment (Pierlou) [inline data_processing/rne/task_functions.py:25]: Je savais pas que cet utilitaire existait 🤯

**Review by Pierlou — APPROVED** [review]

File: data_processing/rne/task_functions.py (hunk 1 above)
Comment (HAEKADI) [reply data_processing/rne/task_functions.py:25]: Je découvre votre code au fir et à mesure aussi 😉

**Review by geoffreyaldebert — APPROVED** [review]

File: data_processing/rne/flux/task_functions.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    last_siren = None  # Initialize last_siren
+
+    json_file_name = f"rne_flux_{start_date}.json"
+    json_file_path = f"{DATADIR}/{json_file_name}"
+
+    if not os.path.exists(DATADIR):
+        logging.info(f"********** Creating {DATADIR}")
+        os.makedirs(DATADIR)
+
+    with open(json_file_path, "w") as json_file:
```
Comment (XavierJp) [inline data_processing/rne/flux/task_functions.py:84]: can the append mode `"a"` be more efficient ?

File: data_processing/rne/flux/task_functions.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+        logging.info(f"****** Opening file: {json_file_path}")
+        while True:
+            url = f"{RNE_API_DIFF_URL}from={start_date}&to={end_date}&pageSize=100"
+
+            if last_siren:
+                url += f"&searchAfter={last_siren}"
+
+            try:
+                r = make_api_request(session, url, auth, headers)
+                page_data = r.json()
```
Comment (XavierJp) [inline data_processing/rne/flux/task_functions.py:94]: `.json()` and token renewal should be inside `make_api_request`

**Review by XavierJp — COMMENTED** [review]
Very nice, will be perfect soon :)

File: data_processing/rne/flux/task_functions.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+    end_date: str,
+    session,
+    auth: List[Dict],
+    token: Union[str, None],
+):
+    if not token:
+        # If no token is provided, fetch a new one
+        logging.info("Getting new token...")
+        token = get_new_token(session, RNE_API_TOKEN_URL, auth)
+
```
[the remark this answers is not in the document]
Comment (HAEKADI) [reply data_processing/rne/flux/task_functions.py:73]: I'm not sure I understand which logic, could you elaborate a bit more? The whole function is related to RNE except when file is sent to MinIO.

File: data_processing/rne/flux/task_functions.py [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+                url += f"&searchAfter={last_siren}"
+
+            try:
+                r = make_api_request(session, url, auth, headers)
+                page_data = r.json()
+
+                last_siren = get_last_siren_in_page(page_data)
+
+                if page_data:
+                    json.dump(page_data, json_file)
```
[the remark this answers is not in the document]
Comment (HAEKADI) [reply data_processing/rne/flux/task_functions.py:99]: Indeed, it's what the dump does.

**Review by XavierJp — APPROVED** [review]
Looks good

Great job on absorbing RNE inherent complexity 👍

**Review by Pierlou — APPROVED** [review]

**Pull request merged**